hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/CompactHelpers.java

```diff
@@ -82,7 +82,7 @@ public HoodieCommitMetadata createCompactionMetadata(
   public void completeInflightCompaction(HoodieTable table, String compactionCommitTime, HoodieCommitMetadata commitMetadata) {
     HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
     try {
-      activeTimeline.transitionCompactionInflightToComplete(
+      activeTimeline.transitionCompactionInflightToComplete(false, table.getConfig().getTimeGeneratorConfig(),
```

Comment:
Are we sure there is no need to lock here?

Reply:
Yes, the outside caller use transitionManager to guarantee this.